src/word.c

```diff
@@ -0,0 +1,67 @@
+#include "dictionary.h"
+#include <stdio.h>
+#include <strings.h>
+#include <string.h>
+#include <assert.h>
+#include <stdlib.h>	
+
+int num_punctuation = 22;
+char* punctuation_array[] = {"+",",",":"," ",".","-","'","&","!","?",":",";","#","~","=","/","$","£","^","\n","_","<",">"};
+
+// checks if punctuation exists within array
+int is_in_array(char* punctuation_array[], char* word) {
+    int i = 0;
+	for ( ; i < num_punctuation ; i++) {
+		if (strcmp(punctuation_array[i], word) == 0) {
+			//printf("comparing %s to %s", word, punctuation_array[i]);
+			return 1;	}
+	}
+	return -1;
+}
+
+// checks if word is valid -- does not have erroneous punctuations within
+int valid_word(char* word, dict_t* dict) {
```

Comment:
The argument ordering is inconsistent with `dict_exists`.

Reply:
fixed